sycl/test-e2e/Matrix/Inputs/joint_matrix_half_impl.hpp

```diff
@@ -63,45 +69,57 @@ void matrix_multiply(big_matrix<TResult, M, N> &C, big_matrix<T, M, K> &A,
                      (k * TK / VNNI) * (N * VNNI) +
                      sg_starty / sg_size * TN * VNNI,
                  N * VNNI);
-             joint_matrix_mad(sg, sub_c, sub_a, sub_b, sub_c);
+             joint_matrix_mad(sg, sub_d, sub_a, sub_b, sub_c);
+             joint_matrix_copy(sg, sub_d, sub_c);
            }
            joint_matrix_store(
-               sg, sub_c,
-               accC.template get_multi_ptr<access::decorated::no>() +
+               sg, sub_d,
+               accD.template get_multi_ptr<access::decorated::no>() +
                    (sg_startx * TM) * N + sg_starty / sg_size * TN,
                N, layout::row_major);
          }); // parallel for
    }).wait();
 }
 
-template <typename TResult, typename T, size_t VNNI, size_t TM, size_t TN,
-          size_t TK>
+template <typename T, typename TAcc, typename TResult, size_t VNNI, size_t TM,
+          size_t TN, size_t TK>
 void test() {
-  static constexpr size_t MATRIX_M = TM * 2;
-  static constexpr size_t MATRIX_N = TN * 2;
-  static constexpr size_t MATRIX_K = TK * 2;
+  std::cout << "Testing: " << TM << " x " << TN << " x " << TK
+            << " [TM x TN x TK]" << std::endl;
+
+  static constexpr size_t MATRIX_M = TM;
+  static constexpr size_t MATRIX_N = TN;
+  static constexpr size_t MATRIX_K = TK;
   T A[MATRIX_M][MATRIX_K];
-  T B[MATRIX_K / VNNI][MATRIX_N * VNNI];
-  TResult C[MATRIX_M][MATRIX_N];
+  T B[MATRIX_K][MATRIX_N];
+  T BVnni[MATRIX_K / VNNI][MATRIX_N * VNNI];
+  TAcc C[MATRIX_M][MATRIX_N];
   TResult D[MATRIX_M][MATRIX_N];
+  TResult DRef[MATRIX_M][MATRIX_N];
 
+  // matrix_rand<T>(MATRIX_M, MATRIX_K, (T *)A, T(1));
+  // matrix_rand<T>(MATRIX_K, MATRIX_N, (T *)B, T(1));
   matrix_fill(MATRIX_M, MATRIX_K, (T *)A,
-              [](int i, int j) { return i + 2 * j; });
-  matrix_fill(MATRIX_K / VNNI, MATRIX_N * VNNI, (T *)B,
-              [](int i, int j) { return i + j; });
-  matrix_fill(MATRIX_M, MATRIX_N, (TResult *)C, TResult(1));
-  matrix_fill(MATRIX_M, MATRIX_N, (TResult *)D, TResult(1));
+                [](int i, int j) { return (i == j) ? T(1) : T(0); });
+  matrix_fill(MATRIX_K, MATRIX_N, (T *)B, [](int i, int j) { return i; });
```

Comment:
Please, use rand or f(i,j) which would return different values for different cols and rows.
Using identity matrix or same value either for rows or cols may hide bugs.